lib/core/src/Cardano/Wallet/Primitive/Model.hs

```diff
@@ -228,19 +217,39 @@ initWallet
     -> BlockchainParameters
         -- ^ Initial blockchain parameters
     -> s
-    -> Wallet s t
+        -- ^ Initial address discovery state
+    -> ([(Tx t, TxMeta)], Wallet s t)
 initWallet block bp s =
     let
-        ((txs, utxo'), s') = prefilterBlock (Proxy @t) block mempty s
-    in Wallet utxo' (Set.fromList txs) (header block) s' bp
+        ((txs, u), s') = prefilterBlock (Proxy @t) block mempty s
+    in
+        (txs, Wallet u (header block) s' bp)
```

Comment:
There was a pretty "big" unnoticed issue here :open_mouth: ! We were putting transactions discovered in the first block as pending transactions for the wallet !!!
Instead, I've chosen to now return these transactions as part of the `initWallet` call (which signature and implementation looks very much like `applyBlock` now... we could use some more refactoring here)

Reply:
awesome this was caught now

Reply:
🙈 👍